### PR TITLE
[2/15] Added comments and docs to the swapchain class

### PR DESCRIPTION
**Pull request by Raelr**
Added comments and TODOs to the Swapchain class.

### CODE REVIEW

File: .github/workflows/macOS.yml [hunk 1]
```diff
@@ -21,7 +21,7 @@ jobs:
 
       - name: build project
         run: |
-          make package DEBUG=0
+          make package ENABLE_VALIDATION_LAYERS=0
```
Comment (jonjondev) [inline .github/workflows/macOS.yml]: why does this need to be explicitly defined?

File: src/Renderer/Swapchain/Swapchain.h [hunk 2]
```diff
@@ -4,58 +4,212 @@
...
 
+        /**
+         * @brief Creates a new Swapchain object to buffer our images. 
+         * 
+         * @param device a reference to a device. Devices are always destroyed last, so we can 
+         * guarantee that object destruction will occur at the correct time. 
+         * @param windowExtent the window resolution. Images need to be scaled to an extent, this usually
+         * must follow the extents of the window. 
+         */
         SwapChain(VulkanDevice& device, VkExtent2D windowExtent);
```
Comment (jonjondev) [inline src/Renderer/Swapchain/Swapchain.h:44]: I remember you said that you'd eventually remove references to devices, so I think this is changed later.

File: src/Renderer/Swapchain/Swapchain.h [hunk 3]
```diff
@@ -4,58 +4,212 @@
...
         ~SwapChain();
 
+        // Delete copy constructors. 
         SwapChain(const SwapChain&) = delete;
         void operator=(const SwapChain*) = delete;
```
Comment (jonjondev) [inline src/Renderer/Swapchain/Swapchain.h:49]: Need to remember to do this on ADD later down the road.

File: src/Renderer/Swapchain/Swapchain.h [hunk 4]
```diff
@@ -4,58 +4,212 @@
...
 
         // Getters
 
+        /**
+         * @brief Images can be formatted in a number of ways. The renderer currently 
+         * searches for a specific format to render to. This function returns the format
+         * currently being used by the rendrer.
```
Comment (jonjondev) [inline src/Renderer/Swapchain/Swapchain.h]:
```suggestion
         * currently being used by the renderer. 
```

File: src/Renderer/Swapchain/Swapchain.h [hunk 5]
```diff
@@ -4,58 +4,212 @@
...
         void CreateDepthResources();
+
+        /**
+         * @brief Creates the swapchain's frame buffers. A frame buffer represents all the data needed to be submitted 
+         * to the GPU in a frame. 
+         */
         void CreateFrameBuffers();
+
+        /**
+         * @brief Inialises all thread synchronisation objects. Vulkan is multi-threaded and submits images
```
Comment (jonjondev) [inline src/Renderer/Swapchain/Swapchain.h]:
```suggestion
         * @brief Initialises all thread synchronisation objects. Vulkan is multi-threaded and submits images
```

Comment (jonjondev) [reply src/Renderer/Swapchain/Swapchain.h]: I think you have spellcheck turned off in your IDE

**Review by jonjondev — APPROVED** [review]
Looks good. Lovingly commented. I'll quickly rebase it on top of `main`.

File: .github/workflows/macOS.yml (hunk 1 above)
Comment (Raelr) [reply .github/workflows/macOS.yml]: ah, you're right. I think I just saw it and replaced the variables.

Comment (jonjondev) [reply .github/workflows/macOS.yml]: I'll remove during the rebase.

File: src/Renderer/Swapchain/Swapchain.h (hunk 5 above)
Comment (Raelr) [reply src/Renderer/Swapchain/Swapchain.h]: yeah you're right. I removed typos because it was giving me issues with variable names.

**Pull request merged**